events/events.go

```diff
@@ -0,0 +1,142 @@
+package events
+
+import (
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/types"
+)
+
+const (
+	NewBlock channelId = 1 + iota
+	BlockValid
+	NewAtx
+	AtxValid
+	NewTx
+	TxValid
+	RewardReceived
+)
+
+// publisher is the event publisher singleton.
+var publisher *EventPublisher
+
+// Publish publishes an event on the pubsub singleton.
+func Publish(event Event) {
+	if publisher != nil {
+		err := publisher.PublishEvent(event)
+		log.Error("pubsub error: %v", err)
+	}
+}
+
+// InitializeEventPubsub initializes the global pubsub broadcaster server
+func InitializeEventPubsub(ur string) {
+	var err error
+	publisher, err = NewEventPublisher(ur)
+	if err != nil {
+		log.Panic("cannot init pubsub: %v", err)
+	}
+}
+
+// EventPublisher is the struct that publishes events to subscribers by topics.
+type EventPublisher struct {
+	Publisher
+}
+
+// Event defines the interface that each message sent by the EventPublisher needs to implemet for it to correctly
+// be routed by topic.
+type Event interface {
+	// getChannel returns the channel on which this message will be published.
+	getChannel() channelId
+}
+
+// NewEventPublisher is a constructor for the event publisher, it received a url string in format of tcp://localhost:56565 to start
+// listening for connections.
+func NewEventPublisher(eventUrl string) (*EventPublisher, error) {
+	p, err := newPublisher(eventUrl)
+	if err != nil {
+		return nil, err
+	}
+	return &EventPublisher{*p}, nil
+}
+
+// PublishEvent publishes the provided event on pubsub infra. It encodes messages using XDR protocol.
+func (p *EventPublisher) PublishEvent(event Event) error {
+	bytes, err := types.InterfaceToBytes(event)
+	if err != nil {
+		return err
+	}
+	return p.publish(event.getChannel(), bytes)
+}
+
+func (p *EventPublisher) Close() error {
+	return p.sock.Close()
+}
+
+type BasicEvent struct {
+	Id byte
+}
+
+type NewBlockEvent struct {
```

Comment:
I'd flip the naming, as the channel ID consts are used internally, you can change their names. 
the usage of the structs, on the other hand, is always. `events.SomeThingEvent{}`. I like more to see `events.SomeThing` since we already know it's event because it's in `events` :)